platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/Migratable.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.nio.file.Path;
+
+/**
+ * This interface provides the mechanism for implementers to define how their data shall be
+ * exported for later import into a new system.
+ * 
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface Migratable {
+
+    /**
+    * Export all migratable data to the specified directory
+    *
+    * @param exportPath The exportPath is is unique path given to the {@link Migratable} 
+    *                   so it can export data without name collisions.
+    *
+    * @return Metadata describing the results of the migration
+    */
+    public MigrationMetadata export(Path exportPath) throws MigrationException;
+    
+    /**
+    * Get a description of the migratable data
+    * @return A short description of the migratable data
+    */
+    public String getDescription();
+    
+    /**
+    * Get the ID of this {@link Migratable}
+    * @return The ID of this {@link Migratable}
+    */
+    public String getId();
+    
+    /**
+    * Get the name of this {@link Migratable}
+    * @return The name of this {@link Migratable}
+    */
+    public String getName();
+
+    /**
+    * Determines if the exported data from this {@link Migratable} is optional or required. 
+    * This can be used by consumers of {@link Migratable} services to determine if the data
+    * of this export is necessary when importing into a new system.
+    * @return Status of whether or not the export is optional
+    */
+    public boolean isOptional();
```

Comment:
Are we sure that this isOptional boolean is needed? I understand that modules like platform are required for the system to run, but is there ever a situation where exporting/importing a configuration would ever *not* be optional? I think it's a bit odd that if someone wants to export, say, just security configurations, that they are also forced to export any modules that aren't marked as not optional. It seems like anything that implements migratable should be optional when exporting or importing.